lms/djangoapps/class_dashboard/dashboard_data.py

```diff
@@ -180,11 +187,29 @@ def get_d3_problem_grade_distrib(course_id):
                         # Construct label to display for this problem
                         label = "P{0}.{1}.{2}".format(c_subsection, c_unit, c_problem)
 
-                        # Only problems in prob_grade_distrib have had a student submission.
-                        if child.location in prob_grade_distrib:
+                        grade_distribution = []
+
+                        try:
+                            problem = client.modules(course_id, child.location)
```

Comment:
Only requesting the module from the client can raise a NotFoundError.  I've moved the grade_distribution request out of the block, although it makes the code a little unnecessarily verbose.  Is it generally considered bad style to have multiple method calls in a try/catch block?  I can see how if someone added exception handling to grade_distribution() and used the NotFoundError it might create a bug in the future, but that doesn't seem very likely, and it is more readable the other way round.  I've taken it out for now, though.